File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        if(n < 10){
            return n % (n+n) == 0;
        }
        string st = to_string(n);
        int s = 0 , p = 1;
        for(char ch : st){
            s+= ch -'0';
            p *= ch -'0';
        }
        return n % (s + p) == 0;
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.kt
class Solution {
    fun checkDivisibility(n: Int): Boolean {
        var s = 0
        var p = 1
        val st = n.toString()
        for(ch in st){
            s += ch - '0'
            p *= ch - '0'
        }
        return n % (s+p) == 0
    }
}